frontend/src/pages/user/home/Lessons/hooks/useLessons.js

```diff
@@ -4,25 +4,33 @@ import { useDispatch, useSelector } from "react-redux";
 
 import { authenticate } from "../../../../../helpers/auth";
 import { getLessons } from "../../../../../helpers/api";
+import { setCurrentLesson } from "../../../../../store/lessonSlice";
 
-export const useLesson = () => {
+export const useLessons = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch();
   const { user } = useSelector((state) => state.currentUser);
   const [lessons, setLessons] = useState([]);
 
   useEffect(() => {
-    authenticate(navigate, dispatch);
+    if (!user.id) authenticate(navigate, dispatch);
     if (!user.id) return;
 
     getLessons(user.id, (data) => {
       setLessons(data.data.lessons);
     });
```

Comment:
- move this function also in reducer, para dali ra ang pag modify sa state. implement a _Redux createAsyncThunk_